backend/src/database/open_content.go

```diff
@@ -68,3 +69,88 @@ func (db *DB) CreateContentActivity(urlString string, activity *models.OpenConte
 		log.Warn("unable to create content activity for url, ", urlString)
 	}
 }
+
+func (db *DB) ToggleLibraryFavorite(contentParams *models.OpenContentParams) (bool, error) {
+	var fav models.LibraryFavorite
+	url := models.OpenContentUrl{}
+	if db.Where("content_url = ?", contentParams.ContentUrl).First(&url).RowsAffected == 0 {
+		url.ContentURL = contentParams.ContentUrl
+		if err := db.Create(&url).Error; err != nil {
+			log.Warn("unable to create content url for activity")
+			return false, newCreateDBError(err, "open_content_urls")
+		}
+	}
+	if db.Model(&models.LibraryFavorite{}).Where("user_id = ? AND content_id = ? AND open_content_url_id = ?", contentParams.UserID, contentParams.ContentID, url.ID).First(&fav).RowsAffected > 0 {
+		if err := db.Delete(&fav).Error; err != nil {
+			return false, newNotFoundDBError(err, "library_favorites")
+		}
+	} else {
+		newFav := models.LibraryFavorite{
+			UserID:                contentParams.UserID,
+			ContentID:             contentParams.ContentID,
+			OpenContentUrlID:      url.ID,
+			Name:                  contentParams.Name,
+			OpenContentProviderID: contentParams.OpenContentProviderID,
+		}
+		if err := db.Create(&newFav).Error; err != nil {
+			return false, newCreateDBError(err, "library_favorites")
+		}
+	}
+	return true, nil
+}
+
+func (db *DB) GetUserFavorites(userID uint) ([]models.CombinedFavorite, error) {
+	var openContentFavorites []models.CombinedFavorite
+	if err := db.Debug().Table("library_favorites fav").
+		Select(`
+            fav.id,
+            fav.name,
+            'library' as type,
+            fav.content_id,
+            lib.image_url as thumbnail_url,
+            ocp.description,
+            NOT lib.visibility_status AS visibility_status,
+            fav.open_content_provider_id,
+            ocp.name AS provider_name,
+            fav.created_at AS created_at
```

Comment:
`Debug()` can be removed, and here you can just select `fav.created_at` w/o the alias